### DISCUSSION TITLE
WildFly distribution plans for the new storage

### DISCUSSION THREAD

**Discussion by sguilhen** [keycloak/keycloak#10189] [Archive]
Currently the WildFly distribution has only support for the concurrent hashmap implementation. The modules for the new JPA and HotRod implementations are missing in this distribution. Besides wiring the new modules we might run into issues such as Liquibase initialization, given that we have extensions in both legacy and new JPA implementations that need to be visible to the module's classloader that is initializing Liquibase (currently model-jpa as the new storage is optional). This might require the extraction of Liquibase's extensions in to a separate module that is wired into the legacy and new JPA modules and makes all extensions visible, for example.

We need to discuss the plans for this distro given the recent shift towards the Quarkus distribution becoming the default one for Keycloak. Do we have plans to add support for the new implementations or will we have it support only the concurrent hashmap implementation that is there today?

**Comment by stianst** [comment]
Quarkus distribution is already the default one for Keycloak. The WildFly distribution will be supported in major/minor releases for only another 3 months or so, with the remaining 3 months only micro releases. That basically means that in around 3 months we plan to delete all existence of the WildFly distribution in the main branch.

So, I would suggest rather than continue looking at the WF distribution that we start with the integration with the Quarkus distribution, and move all work around the new store to Quarkus.

**Reply by sguilhen** [reply]
Thanks for the input, Stian. I was personally already leaning towards focusing on Quarkus only but I had to bring this up to ensure that we won't have plans to integrate the new storages into WildFly.
